src/transforms/dedupe.rs

```diff
@@ -257,12 +258,10 @@ fn build_cache_entry(event: &Event, fields: &FieldMatchConfig) -> CacheEntry {
 
             if let Some(all_fields) = event.as_log().all_fields() {
                 for (field_name, value) in all_fields {
-                    if !fields.contains(&field_name) {
-                        entry.push((
-                            field_name,
-                            type_id_for_value(value),
-                            value.coerce_to_bytes(),
-                        ));
+                    if let Ok(path) = ConfigTargetPath::try_from(field_name) {
```

Comment:
`Config*Path` types should only be used in config structs. You can just use `OwnedTargetPath::try_from(src)` here which is what `ConfigTargetPath` uses internally.

Preferably this wouldn't parse paths at all though. `FieldsIter` (which is what `all_fields` uses) should probably be converted to return path types instead of strings. This doesn't necessarily have to be included in this PR, but it should be done _eventually_.